src/Ampersand/PatchHelper/Helper/Magento2Instance.php

```diff
@@ -119,6 +102,107 @@ public function __construct(string $path)
             $libPath = sanitize_filepath($dirList->getRoot(), $libPath) . '/';
             $this->listOfPathsToLibrarys[$libPath] = $lib;
         }
+
+        $this->prepareThemes();
+    }
+
+    /**
+     * Prepare theme configuration and a list of paths to ignore in the scanning for hyva projects
+     *
+     * @link https://github.com/AmpersandHQ/ampersand-magento2-upgrade-patch-helper/issues/75
+     *
+     * As I understand it the reqs are
+     * - Only include frontend themes that have a Hyva/ root
+     * - We can just ignore file changes to theme files from non Hyva themes, don't need to check them
+     *
+     * @TODO hyva_theme_fallback/general/theme_full_path may require re-adding back in some excluded themes
+     *
+     * @return void
+     */
+    private function prepareThemes()
+    {
+        $themeList = $this->objectManager->get(ThemeList::class);
+        $allFrontendThemes = $themesToIgnore = [];
+        foreach ($themeList as $theme) {
+            switch ($theme->getArea()) {
+                case Area::AREA_FRONTEND:
+                    $allFrontendThemes[$theme->getCode()] = $theme;
+                    break;
+                case Area::AREA_ADMINHTML:
+                    $this->customAdminThemes[] = $theme;
+                    break;
+            }
+        }
+        unset($theme);
+
+        try {
+            /** @var \Magento\Store\Model\StoreManagerInterface $storeManager */
+            $storeManager = $this->objectManager->get(\Magento\Store\Model\StoreManagerInterface::class);
+            /** @var \Magento\Theme\Model\View\Design $design */
+            $design = $this->objectManager->get(\Magento\Theme\Model\View\Design::class);
+
+            $usedFrontendThemes = [];
+            foreach ($storeManager->getStores(false) as $store) {
+                $themeCode = $design->getConfigurationDesignTheme(
```

Comment:
@convenient , I just tested this. I fear, there is something wrong in this check here.
My `$allFrontendThemes` variable looks like this:
```
[Magento/blank]
[Hyva/reset]
[Hyva/default]
[Magento/luma]
[project/theme]
[project/checkoutTheme]
```
whereas the `$themeCode` is just an ID, instead of some code like above.
I debugged into `getConfigurationDesignTheme` and can see, that I'm ending up in
```
$theme = (string) $this->_scopeConfig->getValue(
    self::XML_PATH_THEME_ID,
    ScopeInterface::SCOPE_STORE,
     $store
);
```
, which returns an ID instead of the theme code like expected.
Am I doing something wrong or do I just have a different setup which leads to this issue?

Reply:
Seems the magento function to load the theme does a little post processing of this value to handle the case where it is an id and not a code.

https://github.com/magento/magento2/blob/7c6b6365a3c099509d6f6e6c306cb1821910aab0/app/code/Magento/Theme/Model/Theme/Resolver.php#L57-L63

Pushed c6b138e611851a9ab5849d064360df95fdde59c4 , how does that fare?

Reply:
@convenient , unfortunately, the ID is not given in the current theme collection. See `\Magento\Framework\View\Design\Theme\ThemeList::_prepareConfigurationData`, where the data is set. So `$frontendTheme->getId()` always returns null and `$frontendThemIdToCode[$themeCode]` throws an error, because the array element is not found. 
Could we maybe use the theme collection from `Magento_Theme` or is there a specific reason why you decided to use `Magento_Framework` instead?

Reply:
Introduces in this commit https://github.com/AmpersandHQ/ampersand-magento2-upgrade-patch-helper/commit/fa73df4762c83801a43b03dc3cde9942878e8c31

I think no real reason. We could get the collection from `Magento_Theme` I think.

Reply:
@norgeindian let me know if the latest commit sorts this and i can resolve this conversation 👍

Reply:
@convenient , yes, this can be resolved in my eyes. Just started testing and the themes are now properly given. Thanks